include/deal.II/dofs/dof_accessor.templates.h

```diff
@@ -4300,6 +4300,23 @@ DoFCellAccessor<DoFHandlerType, level_dof_access>::clear_p_coarsen_flag() const
 }
 
 
+
+template <typename DoFHandlerType, bool level_dof_access>
+inline unsigned int
+DoFCellAccessor<DoFHandlerType,
+                level_dof_access>::active_fe_index_after_refinement() const
+{
+  if (p_refine_flag_set())
+    return this->get_dof_handler().get_fe_collection().next_in_hierarchy(
```

Comment:
I would add a sanity check `Assert(!p_coarsen_flag_set(), ExcInternalError());` here.

Reply:
By setting a flag, we check if a flag of the other type has already been set. But you can never have enough sanity checks, right? :)

Reply:
> But you can never have enough sanity checks, right? :)

yes, i don't think it hurts to have one more here 😄